CreeDictionary/CreeDictionary/views.py

```diff
@@ -12,26 +12,44 @@
 # rationale: we don't unit test the views functions, rather, we test them in integration tests with cypress.
 
 
-def index(request, query_string=None, lemma_id=None):  # pragma: no cover
+def lemma_details(request, lemma_text: str = None):  # pragma: no cover
     """
-    homepage with optional initial query or initial lemma. query_string and lemma_id can not both be given
+    lemma detail page. Fall back to search page if no lemma is found or multiple lemmas are found
+
+    :param request: accepts query params `pos` `full_lc` `analysis` `id` to further specify query_string
+    :param lemma_text: the exact form of the lemma (no spell relaxation)
+    """
+    extra_constraints = {
+        k: v for k, v in request.GET.items() if k in {"pos", "lc", "analysis", "id"}
+    }
+    lemma = Wordform.objects.filter(text=lemma_text, is_lemma=True, **extra_constraints)
+    if lemma.count() == 1:
+        lemma = lemma.get()
+        context = {
+            "lemma_id": lemma.id,
+            "lemma": lemma,
+            "paradigm_size": ParadigmSize.BASIC.display_form,
+            "paradigm_tables": lemma.paradigm if lemma else None,
+        }
+        return HttpResponse(render(request, "CreeDictionary/index.html", context))
+    else:
+        return redirect("cree-dictionary-index-with-query", query_string=lemma_text)
```

Comment:
I really like this redirect! 🤩🤩🤩

Could the logic actually be inverted with a [guard clause](https://wiki.c2.com/?GuardClause)?

e.g., 

```python
if lemma.count() != 1:
    # ambiguous results: search for the lemma instead:
    return redirect("cree-dictionary-index-with-query", query_string=lemma_text)

# this is the normal code:
lemma = lemma.get()
context = {...}
return HttpResponse(render(request, "CreeDictionary/index.html", context))
```